pkg/odo/cli/application/application.go

```diff
@@ -197,17 +192,40 @@ var applicationListCmd = &cobra.Command{
 		apps, err := application.ListInProject(client)
 		odoutil.LogErrorAndExit(err, "unable to get list of applications")
 		if len(apps) > 0 {
-			log.Infof("The project '%v' has the following applications:", projectName)
-			tabWriter := tabwriter.NewWriter(os.Stdout, 5, 2, 3, ' ', tabwriter.TabIndent)
-			fmt.Fprintln(tabWriter, "ACTIVE", "\t", "NAME")
-			for _, app := range apps {
-				activeMark := " "
-				if app.Active {
-					activeMark = "*"
+
+			if outputFlag == "json" {
+				var appList []application.App
+				for _, app := range apps {
+					appDef := getMachineReadableFormat(client, app.Name, projectName, app.Active)
+					appList = append(appList, appDef)
+				}
+
+				appListDef := application.AppList{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "List",
+						APIVersion: "odo.openshift.io/v1alpha1",
+					},
+					ListMeta: metav1.ListMeta{},
+					Items:    appList,
 				}
-				fmt.Fprintln(tabWriter, activeMark, "\t", app.Name)
+				out, err := json.Marshal(appListDef)
+				odoutil.LogErrorAndExit(err, "")
+				fmt.Println(string(out))
+
+			} else {
```

Comment:
I would say we should error out if -o is passed but its not json(basically supported formats)
and show normal table format only if -o is not passed

Reply:
Added: https://github.com/redhat-developer/odo/pull/1196/files#diff-743131ef559d43bf35b5ea116eac9d68R48

Reply:
please also do this for all commands

Reply:
once, `complete-validate-run` is merged, I will make change there

Reply:
@surajnarwade are you going to wait for complete-validate-run for application before merging this?
I don't think we should do that. This PR needs to validate that regardless